code/modules/mob/living/carbon/human/species.dm

```diff
@@ -255,50 +256,61 @@ GLOBAL_LIST_EMPTY(features_by_species)
 
 	if(heart && (!should_have_heart || replace_current))
 		heart.Remove(C,1)
+		required_organs -= heart
 		QDEL_NULL(heart)
 	if(should_have_heart && !heart)
 		heart = new mutant_heart()
 		heart.Insert(C)
+		required_organs += heart
```

Comment:
This is still storing atoms in a list and doesn't remove them from that list when the atom is deleted. If you put a heart in this list, the heart **must** be removed when the heart gets deleted which breaks the purpose of this list. You need to store primitive types instead, such as a list of organ slot positions.

Reply:
I think you need to put a typepath manually instead of a variable